src/query/pipeline/core/src/processors/processor.rs

```diff
@@ -102,13 +115,13 @@ impl ProcessorPtr {
     pub fn create(inner: Box<dyn Processor>) -> ProcessorPtr {
         ProcessorPtr {
             id: Arc::new(UnsafeCell::new(node_index(0))),
-            inner: Arc::new(UnsafeCell::new(inner)),
+            inner: Arc::new(UnsafeCell::new(inner).into()),
         }
     }
 
     /// # Safety
     pub unsafe fn as_any(&mut self) -> &mut dyn Any {
-        (*self.inner.get()).as_any()
+        (*self.inner.0.get()).as_any()
```

Comment:
Suggest impl `Deref` for `UnsafeSyncCelledProcessor`, then we don't need add `.0` everywhere.

Reply:
good point, thanks